pkg/bootstrap/platform/aws.go

```diff
@@ -118,16 +125,24 @@ func getEC2MetadataClient() *ec2metadata.EC2Metadata {
 // Note: avoided importing the satellite package directly to reduce number of
 // dependencies, etc., required.
 func systemInfoSuggestsAWS() bool {
-	hypervisorUUIDBytes, _ := os.ReadFile("/sys/hypervisor/uuid")
+	hypervisorUUIDBytes, uerr := os.ReadFile("/sys/hypervisor/uuid")
 	hypervisorUUID := strings.ToLower(string(hypervisorUUIDBytes))
 
-	productUUIDBytes, _ := os.ReadFile("/sys/class/dmi/id/product_uuid")
+	productUUIDBytes, perr := os.ReadFile("/sys/class/dmi/id/product_uuid")
 	productUUID := strings.ToLower(string(productUUIDBytes))
 
 	hasEC2Prefix := strings.HasPrefix(hypervisorUUID, "ec2") || strings.HasPrefix(productUUID, "ec2")
 
-	version, _ := os.ReadFile("/sys/class/dmi/id/product_version")
+	version, verr := os.ReadFile("/sys/class/dmi/id/product_version")
 	hasAmazonProductVersion := strings.Contains(string(version), "amazon")
 
-	return hasEC2Prefix || hasAmazonProductVersion
+	// If the error is a permission error, treat it as AWS as the files exist but user does not have
```

Comment:
These files exist everywhere?
```
$ docker run ubuntu cat /sys/class/dmi/id/product_uuid
9a8ecd0f-ec04-e911-9c8b-f4939fef9664
```

So this seems unsafe.

Reply:
Yeah, I think some of these are checked in the Azure (and GCP maybe?) checks as well.

Reply:
Yes. These are checked already. 

@howardjohn Did not understand the "unsafe" part? Are you saying we should not read them at all? Or some problem with permission checking?

Reply:
I mean this statement is basically always true. So we will always go down the AWS path even outside if AWS?

Reply:
If there are permissions errors  we just skip system info check but still go ahead and check with EC2 metadata https://github.com/istio/istio/blob/master/pkg/bootstrap/platform/aws.go#L42 which will fail if it is not AWS. Without this if there is a permission error on a real AWS, it just thinks it is not AWS.

Reply:
I think the extra check is because if you have restrictive network policies the EC2 metadata check may hang forever (except we have a timeout, but its not very low). My memory here may be wrong though @douglas-reid ?

Reply:
if EC2 metadata can not be accessed, even if it is detected as AWS based on sys files - we wont get locality info also ?

Reply:
I mean if you call getEC2MetadataClient/client.Available off-AWS with restrictive network policies it may hang forever. So this may break non-AWS users

Reply:
This was an issue we had with off-GCP testing, iirc. Slow detection leading to long(er) proxy start times. The extra check was meant to be a rough cut short-circuit for off-platform users. If we don't have file perms, however, i'm not sure what choice we have. Should we have a `other` or `none` option for the env var that skips platform detection entirely to allow for quicker startups in those cases?

Reply:
> The extra check was meant to be a rough cut short-circuit for off-platform users. If we don't have file perms, however, i'm not sure what choice we have.

Yes. Exactly. That is why I felt we could take a chance and call AWS api if file permissions are not there - then it would be slow but people would be fine if it detects correctly. And even if we have file permissions and it detects AWS, does not the call to metadata for locality would run in to same slowness?

> Should we have a other or none option for the env var that skips platform detection entirely to allow for quicker startups in those cases?

We can have and in this case where would it get locality from ? Is it from `istio-locality` label?

Reply:
RE: locality. `istio-locality` seems like the right place, I guess. that should work, right?

Reply:
Did some brief testing and restrictive network policy already takes 5s before this PR anyways.

So IMO this is ok. We can/should add NONE to allow skipping. Its valid to be "none" - locality info is not detected in the proxy but can be grabbed from the nodes.